code/modules/mining/equipment/kinetic_crusher/trophies/lavaland_megafauna_trophies.dm

```diff
@@ -0,0 +1,183 @@
+//Lavaland megafauna trophies go here.
+
+
+/**
+ * Blood-drunk miner
+ * Detonating a mark applies a blood-drunk status effect to the user, absorbing stun effects and
+ * reducing ALL incoming damage by 90% for 1 second.
+ */
+/obj/item/crusher_trophy/miner_eye
+	name = "eye of a blood-drunk hunter"
+	desc = "Its pupil is collapsed and turned to mush. Suitable as a trophy for a kinetic crusher."
+	icon_state = "hunter_eye"
+	denied_types = list(/obj/item/crusher_trophy/miner_eye)
+
+/obj/item/crusher_trophy/miner_eye/effect_desc()
+	return "mark detonation to grant stun immunity and <b>90%</b> damage reduction for <b>1</b> second"
+
+/obj/item/crusher_trophy/miner_eye/on_mark_detonation(datum/source, mob/living/target, mob/living/user)
+	. = ..()
+
+	user.apply_status_effect(/datum/status_effect/blooddrunk)
+
+/**
+ * Ash drake
+ * Detonating a mark causes any enemies near the user in a 2-tile radius to take fire damage
+ * and be pushed away a tile, including the initial mark victim.
+ * Any mobs regardless of faction are affected.
+ */
+/obj/item/crusher_trophy/tail_spike
+	desc = "A spike taken from an ash drake's tail. Suitable as a trophy for a kinetic crusher."
+	denied_types = list(/obj/item/crusher_trophy/tail_spike)
+	///How much damage does the effect do to an affected mob
+	var/blast_damage = 5
+
+/obj/item/crusher_trophy/tail_spike/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/crusher_damage_ticker, APPLY_WITH_SPELL, blast_damage)
+
+/obj/item/crusher_trophy/tail_spike/effect_desc()
+	return "mark detonation to do <b>[blast_damage]</b> damage to nearby creatures and push them back"
+
+/obj/item/crusher_trophy/tail_spike/on_mark_detonation(datum/source, mob/living/target, mob/living/user)
+	. = ..()
+
+	for(var/mob/living/victim in oview(2, user))
+		if(victim.stat == DEAD || victim.faction_check_mob(user))
+			continue
+		playsound(victim, 'sound/magic/fireball.ogg', 20, TRUE)
+		new /obj/effect/temp_visual/fire(get_turf(victim))
+		addtimer(CALLBACK(src, PROC_REF(pushback), victim, user), 1) //no free backstabs, we push AFTER module stuff is done
+		log_combat(user, target, "repelled with a trophy ", src, "crusher damage")
+		victim.adjustFireLoss(blast_damage, forced = TRUE)
+		SEND_SIGNAL(src, COMSIG_CRUSHER_SPELL_HIT, victim, user, blast_damage)
```

Comment:
Make a proc `/obj/item/crusher_trophy/report_spell_damage` or some such and send the signal from there, calling it from each trophy with one of these effects. It's better than sending the same signal from a bunch of different procs.

What you do for the tentacles and hierophant chaser... 🤷 
Maybe you can think of an alternate solution. Perhaps you can just pass an appropriate existing triggering signal into the component as an argument?

Reply:
Or register a signal on the spawned effect from the trophy when it is spawned, which calls `report_spell_damage` when an appropriate signal is sent by the effect.